### PR TITLE
[DOCS] [PUT DFA] Documents inline the child params of source and dest

### PR DESCRIPTION
**Pull request by szabosteve**
Related to https://github.com/elastic/ml-team/issues/187#issuecomment-521739017.

This PR adds the child parameters of `source` and `dest` to the parameters' documentation inline.

### CODE REVIEW

**Review by sophiec20 — CHANGES_REQUESTED** [review]
LGTM for the create job page.

For this PR would also be relevant to add to the definition page for `source`, `dest` and `analyzed_fields` (if the object is small, describe it inline).
https://www.elastic.co/guide/en/elasticsearch/reference/current/ml-dfanalytics-resources.html

File: docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc [hunk 1]
```diff
@@ -62,14 +66,20 @@ and mappings.
...
+  For example, all the numeric fields for {oldetection}.
   
 `dest`::
   (Required, object) The destination configuration, consisting of `index` and 
-  optionally `results_field` (`ml` by default). See 
-  <<ml-dfanalytics-properties,{dfanalytics} properties>>.
+  optionally `results_field` (`ml` by default).
+  
+    `index`:::
+      (Required, string) The _destination index_ for the {dfanalytics}.
```
Comment (karenzone) [inline docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc]: The definition list format does not seem to respect indentation, and the list items are presented at the same level as their parent. 
Screen Shot 2019-08-16 at 9 12 31 PM

You have three colons on line 77, and that one appears to indent correctly.  I tested that approach with the other values, and they lined up correctly, too.  I couldn't find any formal documentation that supports this approach, but it seems to work at least on my local build.  I'll retest after you incorporate the other review comments.

**Review by karenzone — COMMENTED** [review]
I like the way you're keeping the information together rather than sending the user off with a link. Nice!
 
I made a few comments inline about indents.

File: docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc (hunk 1 above)
Comment (szabosteve) [reply docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc]: @karenzone Thank you for this, I went with three colons. According to my local build and the Jenkins test, this change solves the indentation issue.

**Comment by szabosteve** [comment]
@sophiec20 Thank you for your comment. ae6f5a3 incorporates the changes you requested.

**Review by sophiec20 — APPROVED** [review]
LGTM

**Comment by szabosteve** [comment]
Backported to 7.x and 7.3.

**Pull request merged**